src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlActivitySourceHelper.cs

```diff
@@ -20,41 +20,72 @@ internal sealed class SqlActivitySourceHelper
     public static readonly AssemblyName AssemblyName = Assembly.GetName();
     public static readonly string ActivitySourceName = AssemblyName.Name!;
     public static readonly ActivitySource ActivitySource = new(ActivitySourceName, Assembly.GetPackageVersion());
-    public static readonly string ActivityName = ActivitySourceName + ".Execute";
 
-    public static readonly IEnumerable<KeyValuePair<string, object?>> CreationTags = new[]
+    public static TagList GetTagListFromConnectionInfo(string? dataSource, string? databaseName, SqlClientTraceInstrumentationOptions options, out string activityName)
```

Comment:
`TagList` is a sizeable `struct`. It would probably be faster to return by reference a la:

```diff
-public static TagList GetTagListFromConnectionInfo(
+public static void GetTagListFromConnectionInfo(
    string? dataSource,
    string? databaseName,
    SqlClientTraceInstrumentationOptions options,
-   out string activityName)
+   out string activityName,
+   out TagList tags)
```

Probably the name sucks too 🤣 `GetSamplerTagsFromConnectionInfo` 🤷

Reply:
> Probably the name sucks too 🤣 GetSamplerTagsFromConnectionInfo 

Yea I agree. I'm gonna leave it sucky for now cause I don't know what to call it yet. This code may continue to be refactored a bit once I integrate metric instrumentation which will also need what this method is doing.